### DISCUSSION TITLE
NEXRAD Non-Reflectivity Values

### DISCUSSION THREAD

**Discussion by mgrover1** [ARM-DOE/pyart#1334] [Q&A] [answered]
Building on the discussion here - https://openradar.discourse.group/t/reflectivity-and-velocity-resolution-from-aws/97

The main question is whether we are parsing NEXRAD Level 2 files with the proper digital resolution.

Reflectivity should be at 8 bit resolution (every 0.5 dBZ), but other fields such as radial velocity and spectrum width should be higher? With differences in values on the order of 0.1 m/s, etc.

@dopplershift do you know where to find this information so we can verify we are treating this properly? Or do you know if the current implementation in MetPy/Py-ART handle this properly?

**Comment by dopplershift** [comment, marked as answer]
So different fields within the Level2 data do have different precisions and amounts of bits actually used. That information is in [this ICD](https://www.roc.noaa.gov/wsr88d/WSR88D/PublicDocs/ICDs/2620002V.pdf) (which seems to be 404-ing on their site at the moment):

<img width="1083" alt="image" src="https://user-images.githubusercontent.com/221526/203395696-50a7070f-490d-4da1-9552-83fab92ce1c2.png">

So PhiDP and ZDR are stored in 16-bit ints, having 10 and 11 bits of precision, respectively. From the post you linked, I also think there is some confusion. Level 2 vel and reflectivity have 8-bits and due to the limited range, that corresponds to 0.5 increments--which is represented exactly in floating point. Correlation coefficient appears to have more digits, but it's still just 8-bit data.

#903 was trying to accommodate properly for the different data word sizes. I know MetPy gets this right.

**Reply by huihuojia** [reply]
In this table, what is the difference between accuracy and precision?

**Reply by dopplershift** [reply]
I assume it's the prototypical [scientific difference](https://en.wikipedia.org/wiki/Accuracy_and_precision) between accuracy and precision.

**Reply by huihuojia** [reply]
Then take the reflectivity for example, all numbers after the decimal point are basically useless?

**Reply by dopplershift** [reply]
No. The accuracy means that the data are within +/- 1.0 dB of the true value of reflectivity in the atmosphere. What you're talking about is the precision, which in that case is 0.5 dB.

**Reply by scollis** [reply]
This is a great discussion. Suggest taking it to Discorse https://openradar.discourse.group/

**Reply by huihuojia** [reply]
> No. The accuracy means that the data are within +/- 1.0 dB of the true value of reflectivity in the atmosphere. What you're talking about is the precision, which in that case is 0.5 dB.

Say there are two observations, 5 and 5.5 dBZ. For the 5 dBZ observation, the possible truth value is [4.0, 6.0], and for the 5.5 dBZ observation, the possible truth value is [4.5, 6.5]. There are overlaps of the possible truth values for these two observations.

**Reply by dopplershift** [reply]
Yes, that's a true statement.

Per @scollis 's request, though, this is my last reply here. I'll also state that if you really have further questions or concerns regarding that document, those are probably best directed to the [National Weather Service Radar Operations Center](https://www.roc.noaa.gov/WSR88D/), since they're responsible for the design and operation of NEXRAD, as well as the documentation that contains the table in question.

**Comment by mgrover1** [comment]
Thanks @dopplershift !! I spent quite a bit of time trying to track down that table... 

And since that is the case with the different bit percisions, then Py-ART gets this right too (since we follow the same approach y'all did)

Thanks for the quick response here. I am going to repost this on the Open Radar discourse as well to respond to Brenda's discussion.
